File: College Stationary Shop/geektrust.py
from sys import argv
from src import *

def main():
    """
    Sample code to read inputs from the file
    """
    if len(argv) != 2:
        raise Exception("File path not entered")
    file_path = argv[1]

    builder = product_builder.Builder()

    with open(file_path, 'r') as f:
        for row in f.readlines():
            row = row.replace("\n", "")
            columns = row.split(' ')

            if columns[0].startswith("ADD_ITEM"):
                builder.add_item(columns[1], int(columns[2]))

            else:
                builder.print_bill()
    
if __name__ == "__main__":
    main()

File: College Stationary Shop Node/geektrust.js
// import {Builder} from "./src/product_builder.js"
const product_builder = require("./src/product_builder.js")
const fs = require("fs")

const filename = process.argv[2]

fs.readFile(filename, "utf8", (err, data) => {
    if (err) throw err
    var inputLines = data.toString().split("\n")
    builder = new product_builder.Builder();

    for(let i = 0; i < inputLines.length; i++) {
        inputLines[i] = inputLines[i].replace("\r", '');
        line = inputLines[i].split(' ');
        if(line[0] == "ADD_ITEM") {
            builder.add_item(line[1], parseFloat(line[2]));
        } else {
            builder.print_bill();
        }
    }
})
